service/keys/keys_test.go

```diff
@@ -109,3 +109,35 @@ func TestGenerateNewMnemonic(t *testing.T) {
 
 	assert.Equal(t, encryptedChannelsBackupKey.String(), derivedKeyFromKeys.String())
 }
+
+func TestGenerateSwapMnemonic(t *testing.T) {
+	logger.Init(strconv.Itoa(int(logrus.DebugLevel)))
+	gormDb, err := db.NewDB(t)
+	require.NoError(t, err)
+	defer db.CloseDB(gormDb)
+
+	mnemonic := "thought turkey ask pottery head say catalog desk pledge elbow naive mimic"
+	unlockPassword := "123"
+
+	config, err := config.NewConfig(&config.AppConfig{}, gormDb)
+	require.NoError(t, err)
+	config.SetUpdate("Mnemonic", mnemonic, unlockPassword)
+
+	keys := NewKeys()
+	err = keys.Init(config, unlockPassword)
+	require.NoError(t, err)
+
+	mnemonicFromConfig, err := config.Get("Mnemonic", unlockPassword)
+	require.NoError(t, err)
+	require.Equal(t, mnemonic, mnemonicFromConfig)
+
+	seed := bip39.NewSeed(mnemonicFromConfig, "")
+	masterKey, err := bip32.NewMasterKey(seed)
+	require.NoError(t, err)
+
+	swapMnemonic, err := keys.GetSwapMnemonic(masterKey)
+	require.NoError(t, err)
+
+	expectedSwapMnemonic := "truth cargo pluck prefer mosquito symptom review kitchen exile fit corn vault"
```

Comment:
We could add a comment: this matches https://iancoleman.io/bip39/ -> check "Show BIP85" and set BIP85 Index to 128260